extensions/oidc-client-registration/runtime/src/main/java/io/quarkus/oidc/client/registration/OidcClientRegistrationConfigBuilder.java

```diff
@@ -0,0 +1,230 @@
+            this.registrationEnabled = builder.registrationEnabled;
+            this.registerEarly = builder.registerEarly;
+            this.initialToken = builder.initialToken;
+            this.metadata = builder.metadata;
+        }
+
+        @Override
+        public Optional<String> id() {
+            return id;
+        }
+
+        @Override
+        public boolean registrationEnabled() {
+            return registrationEnabled;
+        }
+
+        @Override
+        public boolean registerEarly() {
+            return registerEarly;
+        }
+
+        @Override
+        public Optional<String> initialToken() {
+            return initialToken;
+        }
+
+        @Override
+        public Metadata metadata() {
+            return metadata;
+        }
+    }
+
+    private Optional<String> id;
+    private boolean registrationEnabled;
+    private boolean registerEarly;
+    private Optional<String> initialToken;
+    private OidcClientRegistrationConfig.Metadata metadata;
+
+    OidcClientRegistrationConfigBuilder(OidcClientRegistrationConfig config) {
+        super(config);
+        this.id = config.id();
+        this.registrationEnabled = config.registrationEnabled();
+        this.registerEarly = config.registerEarly();
+        this.initialToken = config.initialToken();
+        this.metadata = config.metadata();
+    }
+
+    @Override
+    protected OidcClientRegistrationConfigBuilder getBuilder() {
+        return this;
+    }
+
+    /**
+     * @param id {@link OidcClientRegistrationConfig#id()}
+     * @return this builder
+     */
+    public OidcClientRegistrationConfigBuilder id(String id) {
+        this.id = Optional.ofNullable(id);
+        return this;
+    }
+
+    /**
+     * @param registrationEnabled {@link OidcClientRegistrationConfig#registrationEnabled()}
+     * @return this builder
+     */
+    public OidcClientRegistrationConfigBuilder registrationEnabled(boolean registrationEnabled) {
+        this.registrationEnabled = registrationEnabled;
+        return this;
+    }
+
+    /**
+     * @param registerEarly {@link OidcClientRegistrationConfig#registerEarly()}
+     * @return this builder
+     */
+    public OidcClientRegistrationConfigBuilder registerEarly(boolean registerEarly) {
+        this.registerEarly = registerEarly;
+        return this;
+    }
+
+    /**
+     * @param initialToken {@link OidcClientRegistrationConfig#initialToken()}
+     * @return this builder
+     */
+    public OidcClientRegistrationConfigBuilder initialToken(String initialToken) {
+        this.initialToken = Optional.ofNullable(initialToken);
+        return this;
+    }
+
+    /**
+     * @param clientName {@link OidcClientRegistrationConfig.Metadata#clientName()}
+     * @param redirectUri {@link OidcClientRegistrationConfig.Metadata#redirectUri()}
+     * @return this builder
+     */
+    public OidcClientRegistrationConfigBuilder metadata(String clientName, String redirectUri) {
```

Comment:
Hi @michalvavrik, We now have a dedicated `Metadata` builder which is good, and this is a good shortcut, I propose to have one more `metadata(String redirectUri)` and that should... and we can follow the same pattern for `OidcTenantConfig`...

There is something I'd like to explore....

I'd like to type something like the following, which might be not too bad for bigger config mappings, for example:
```
 OidcClientRegistrationConfig clientRegConfig = OidcClientRegistrationConfig
                    .authServerUrl(authServerUrl)
                    .metadata()
                        .name("My Client")
                        .redirectUri("https://localhostL8443/redirect")
                        .build();
                    .build();
``` 

I.e, we introduce a convention that a no arg sub-ConfigMapping like `metadata()` creates a builder which returns a variation of `MetadataBuilder`, which, once its `build()` is called will return `OidcClientRegistrationConfigBuilder`, for example, let's call it `OidcClientRegistrationConfigMetadataBuilder`...

So users will have an option to build the whole instance in one structured sequence, for example,
```
OidcTenantConfig build() {
    return 
    OidcTenantConfig.authServerUrl("someurl")
        .tenanId("mytenant")
        .authentication()
            .a("aaa")
            .b("bbb")
            .build)
        .token()
            .a("aaa")
            .b("bbb")
            .build()
     .build();
}
```

When users have to deal with many sub ConnfigMappings and each ConfigMapping can have 5-10 properties, the above can save users some space in their code and be quite optimal...

Reply:
Hey @sberyozkin , I'll do it myself. I just hope I get you right. I am starting on it right now, but for context, this is how I understand requests.

1. add `metadata(String redirectUri)` - I am on it
2. `I'd like to type something like the following, which might be not too bad for bigger config mappings, for example:` but you can do that, I am not certain if I understand you. Please have a look here and point out based on that example what you would like to change. I'll do it.

- https://github.com/quarkusio/quarkus/blob/09aa6c30aa5eb4399fcce36c3ecb62dcd55593f0/extensions/oidc-client-registration/runtime/src/test/java/io/quarkus/oidc/client/registration/OidcClientRegistrationConfigBuilderTest.java#L71
- https://github.com/quarkusio/quarkus/blob/09aa6c30aa5eb4399fcce36c3ecb62dcd55593f0/extensions/oidc-client-registration/runtime/src/test/java/io/quarkus/oidc/client/registration/OidcClientRegistrationConfigBuilderTest.java#L161
- https://github.com/quarkusio/quarkus/blob/09aa6c30aa5eb4399fcce36c3ecb62dcd55593f0/extensions/oidc-client-registration/runtime/src/test/java/io/quarkus/oidc/client/registration/OidcClientRegistrationConfigBuilderTest.java#L183
- https://github.com/quarkusio/quarkus/blob/09aa6c30aa5eb4399fcce36c3ecb62dcd55593f0/extensions/oidc-client-registration/runtime/src/test/java/io/quarkus/oidc/client/registration/OidcClientRegistrationConfigBuilderTest.java#L256
3. `So users will have an option to build the whole instance in one structured sequence, for example,` I still think that is what is done here. Only difference is that I can see is that I differentiate between `end` and `build`. But how else to do it?

Reply:
Done, I have added `metadata(String redirectUri)` and test for it.

Reply:
@michalvavrik Oh it is great, I missed `.end()` for a metadata() sub-builder...